### PR TITLE
Improve performance of `is_sink` predicate (and some clean up)

### PR DESCRIPTION
**Pull request by SSoelvsten**

### CODE REVIEW

File: src/adiar/zdd/pred.cpp [hunk 1]
```diff
@@ -156,7 +156,7 @@ namespace adiar {
   bool zdd_disjoint(const zdd &s1, const zdd &s2)
   {
     if (s1.file._file_ptr == s2.file._file_ptr) {
-      return is_sink(s1, is_false);
+      return is_sink(s1) && !value_of(s1);
```
Comment (AnnaBlume99) [inline src/adiar/zdd/pred.cpp:159]: You could still have (inline) functions is_true_sink and is_false_sink that do this kind of work to make it look nicer/a bit more readable.

File: src/adiar/zdd/subset.cpp [hunk 2]
```diff
@@ -162,8 +162,8 @@ namespace adiar
 
   __zdd zdd_onset(const zdd &dd, const label_file &l)
   {
-    if (l.size() == 0 || is_sink(dd, is_false)) { return dd; }
-    if (is_sink(dd, is_true) || disjoint_labels<label_file, label_stream<>>(l, dd)) {
+    if (l.size() == 0 || (is_sink(dd) && !value_of(dd))) { return dd; }
```
Comment (AnnaBlume99) [inline src/adiar/zdd/subset.cpp:165]: Here a is_false_sink function could also be used (see other comment)

File: src/adiar/zdd/subset.cpp [hunk 3]
```diff
@@ -162,8 +162,8 @@ namespace adiar
...
+    if ((is_sink(dd) && value_of(dd)) || disjoint_labels<label_file, label_stream<>>(l, dd)) {
```
Comment (AnnaBlume99) [inline src/adiar/zdd/subset.cpp:166]: Here a is_true_sink function could also be used (see other comment)

File: example/queens.cpp [hunk 4]
```diff
@@ -264,7 +264,7 @@ uint64_t n_queens_list(uint64_t N, uint64_t column,
                        std::vector<uint64_t>& partial_assignment,
                        const adiar::bdd& constraints)
 {
-  if (adiar::is_sink(constraints, adiar::is_false)) {
+  if (adiar::is_sink(constraints) && !adiar::value_of(constraints)) {
```
Comment (AnnaBlume99) [inline example/queens.cpp:267]: Here a is_false_sink function could also be used (see other comment)

File: example/queens.cpp [hunk 5]
```diff
@@ -327,7 +327,7 @@ uint64_t n_queens_list(uint64_t N, uint64_t column,
       for (uint64_t c = N-1; c > column; c--) {
         partial_assignment.pop_back();
       }
-    } else if (adiar::is_sink(restricted_constraints, adiar::is_true)) {
+    } else if (adiar::is_sink(restricted_constraints) && adiar::value_of(restricted_constraints)) {
```
Comment (AnnaBlume99) [inline example/queens.cpp:330]: Here a is_true_sink function could also be used (see other comment)

**Review by AnnaBlume99 — APPROVED** [review]
There are a few places where having is_false_sink and is_true_sink functions might make the code nicer to read, but it is a very small thing.

File: src/adiar/zdd.h [hunk 6]
```diff
@@ -219,6 +219,22 @@ namespace adiar
...
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  /// \brief Whether a given decision diagram represents the true sink.
+  //////////////////////////////////////////////////////////////////////////////
+  inline bool is_null(const decision_diagram &dd)
+  {
+    return is_true(dd);
+  }
+
```
Comment (SSoelvsten) [inline src/adiar/zdd.h:237]: The documentation comments should be set-related.

**Pull request merged**